src/EntityComponentManager.cc

```diff
@@ -1283,6 +1280,8 @@ void EntityComponentManager::SetState(
         // When above TODO is addressed, uncomment AddModifiedComponent below
         // unless calling SetChanged (which already calls AddModifiedComponent)
         // this->dataPtr->AddModifiedComponent(entity);
+        comp->Deserialize(istr);
+        this->SetChanged(entity, typeId, ComponentState::PeriodicChange);
```

Comment:
I believe this fix is not part strictly to the async implementation, should this be targeted to ign-gazebo in another PR? @mjcarroll

Reply:
I agree, this seems unrelated?

Reply:
At first I thought it was unrelated, but it seems the added remove component didn't allow the code to get until that part. So I think now they make sense together. Let me know WDYT